framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -462,6 +462,20 @@ private void applyDefaults(Tree tree, AnnotatedTypeMirror type) {
                 // (The above probably means that we should use defaults in the
                 // scope of the declaration of the array.  Is that right?  -MDE)
 
+            case CLASS:
+                if (((ClassTree) tree).getExtendsClause() != null) {
+                    elt = TreeUtils.elementFromTree(((ClassTree) tree).getExtendsClause());
+                    applyDefaultsToExtendsAndImplicits(elt, type);
+                }
+                if (!((ClassTree) tree).getImplementsClause().isEmpty()) {
+                    List<? extends Tree> implicitClauses = ((ClassTree) tree).getImplementsClause();
+                    for (Tree implicitClause : implicitClauses) {
+                        elt = TreeUtils.elementFromTree(implicitClause);
+                        applyDefaultsToExtendsAndImplicits(elt, type);
+                    }
+                }
+                elt = nearestEnclosingExceptLocal(tree);
```

Comment:
This unconditionally resets `elt`, which may have been set just above.  I suspect the uses just above should be fresh local variables, to avoid confusion about what they are for and their (non-)relationship to the point of the switch statement, which is to set the `elt` local variable.
To clarify the logic of the `applyDefaults` method, how about abstracting its first half (the switch statement) into a helper method?